### PR TITLE
fix: use global() to get performance API on wasm32-unknown-unknown

### PR DESCRIPTION
**Pull request by Officeyutong**
Currently quanta uses `window.performance` to access Performance API in browser, which will not work in WebWorkers. In WebWorkers, there is no `window` object, but `performance` is in the global space.

This PR uses `web_sys::js_sys::global()` to access the global scope, and look up for `performance` in that, which will both work on webpage and WebWorkers.

See https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Functions_and_classes_available_to_workers for more details

### CODE REVIEW

**Review by tobz — COMMENTED** [review]
Thanks for opening this PR!

Taking a look at the code, it seems like this is going to be somewhat inefficient, reflecting through the global scope on every single call.

Is my intuition wrong here? Could we not instead simply query for the `Performance` class when constructing `Monotonic` so that there's always a live reference to it to directly use?

**Comment by Officeyutong** [comment]
> Thanks for opening this PR!
> 
> Taking a look at the code, it seems like this is going to be somewhat inefficient, reflecting through the global scope on every single call.
> 
> Is my intuition wrong here? Could we not instead simply query for the `Performance` class when constructing `Monotonic` so that there's always a live reference to it to directly use?

You are right, Reflection is relatively slow since its equal to `object[attribute]` in JS. I'll try to store a reference to `Performance` class to avoid frequent call to Reflection

**Comment by Officeyutong** [comment]
> Thanks for opening this PR!
> 
> Taking a look at the code, it seems like this is going to be somewhat inefficient, reflecting through the global scope on every single call.
> 
> Is my intuition wrong here? Could we not instead simply query for the `Performance` class when constructing `Monotonic` so that there's always a live reference to it to directly use?

Updated now. I also splitted `wasm.rs` to `wasm_browser.rs` and `wasm_wasi.rs`, since implementation of these two platforms are completely different

**Comment by Officeyutong** [comment]
Hi @tobz , how's this issue going on?

**Review by tobz — CHANGES_REQUESTED** [review]
Apologies: life and work have been pretty busy lately. :)

I left a common about how we caching the `Performance` object, and in particular about how we're overriding the fact that it's `!Send`/`!Sync`.

File: Cargo.toml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name = "quanta"
 version = "0.12.3"
 authors = ["Toby Lawrence <toby@nuclearfurnace.com>"]
 edition = "2021"
-rust-version = "1.60"
+rust-version = "1.70"
```
Comment (tobz) [inline Cargo.toml:6]: I just realized we were changing the MSRV here, but it's not clear why.

Comment (Officeyutong) [reply Cargo.toml:6]: `OnceCell` was stable since `1.70`, see https://doc.rust-lang.org/stable/src/core/cell/once.rs.html#31

Comment (tobz) [reply Cargo.toml:6]: Ah right, makes sense.

**Review by tobz — APPROVED** [review]
Overall, these changes seem reasonable to me. 👍🏻

**Comment by Officeyutong** [comment]
> Overall, these changes seem reasonable to me. 👍🏻

Hi, when can we get this PR merged

**Comment by tobz** [comment]
👋🏻 

Going to merge these now, but I'll have to fix the MSRV and CI afterwards and make sure things are still green before I can cut a release.

**Comment by tobz** [comment]
Released as `quanta@v0.12.4`.

Thanks again for your contribution!

**Pull request merged**